Add flexbox to cipher list to correct display center issue.

gryffs opened the pull request:
This is a better way to correct the issue in #599 
First pull request I made for this issue was #855 But I closed it because it had side effects.

I've added an additional `<a>` wrapper to implement better flexbox functionality so that the items in question can consistency be centered. This also required adding an additional scss field fo referenced wrapper.

Before: 
<img width="468" alt="Screen Shot 2021-05-03 at 4 33 30 PM" src="https://user-images.githubusercontent.com/16548994/116860569-59373800-ac34-11eb-8afc-b7c9d620660d.png"><img width="245" alt="Screen Shot 2021-05-03 at 4 34 13 PM" src="https://user-images.githubusercontent.com/16548994/116860603-6522fa00-ac34-11eb-8b30-82b1168d0c80.png">

After:
<img width="448" alt="Screen Shot 2021-05-03 at 4 39 29 PM" src="https://user-images.githubusercontent.com/16548994/116860685-84ba2280-ac34-11eb-9991-26c6f26bfb8d.png"><img width="247" alt="Screen Shot 2021-05-03 at 4 39 42 PM" src="https://user-images.githubusercontent.com/16548994/116860700-8b489a00-ac34-11eb-9dca-054ee8dc8ada.png">

I've confirmed that the side effects mentioned in the previous pull request have been corrected. The "After" screenshot above shows the list display of all types (I could not add an attachment to confirm placement, but share notification is there).

Let me know what you think, hope this helps.

cscharf commented on src/app/vault/ciphers.component.html:
```diff
@@ -12,20 +12,22 @@
-                    </ng-container>
-                </span>
-                <span class="detail">{{c.subTitle}}</span>
+                <a class='flex-cipher-list-item'>
```
Please use double quotes to encapsulate HTML attributes.

cscharf replied:
Also, does the `a > a` structure mess with the tab index or keyboard interaction (accessibility)? Just wondering if it wouldn't be more semantically correct to use a `<span>` or since HTML5 allows it, a `<div>`? No particular prejudice against `<a>` here, just want to ensure it's interactive via keyboard still.

gryffs replied:
Thanks and I see your point. I would probably opt for the `<div>` since `<span>` has a history of `display:inline`. Keyboard interaction looks to remain intact and accessibility is functioning the same as before as well.
Any thoughts around that before I push up these updates?

Thanks again for the feedback.

cscharf replied:
Let's go with `<div>`, thanks!

cscharf commented:
Thanks @gryffs , looks good, just a couple of comments (one small syntax change and one question).

gryffs commented:
Corrected a syntax issue and changed the above referenced `<a>` element to a `<div>` element. Also attached a small video showing tab through on list.


https://user-images.githubusercontent.com/16548994/116944228-a7d1ea00-aca7-11eb-821e-33cae61f930e.mov

cscharf reviewed (approved):
Thanks!
